coalib/bearlib/abstractions/Linter.py

```diff
@@ -0,0 +1,680 @@
+
+                try:
+                    groups["severity"] = severity_map[
+                        groups["severity"].lower()]
+                except KeyError:
+                    self.warn(
+                        "No correspondence for " + repr(groups["severity"]) +
+                        " found in given severity map. Assuming "
+                        "`RESULT_SEVERITY.NORMAL`.")
+                    groups["severity"] = RESULT_SEVERITY.NORMAL
+
+                for variable in ("line", "column", "end_line", "end_column"):
+                    groups[variable] = (None
+                                        if groups.get(variable, "") == "" else
+                                        int(groups[variable]))
+
+                if "origin" in groups:
+                    groups["origin"] = "{} ({})".format(
+                        str(klass.__name__),
+                        str(groups["origin"]))
+
+                # Construct the result.
+                return Result.from_values(
+                    origin=groups.get("origin", self),
+                    message=groups.get("message", ""),
+                    file=filename,
+                    severity=int(groups.get("severity",
+                                            RESULT_SEVERITY.NORMAL)),
+                    line=groups["line"],
+                    column=groups["column"],
+                    end_line=groups["end_line"],
+                    end_column=groups["end_column"])
+
+            def process_output_corrected(self,
+                                         output,
+                                         filename,
+                                         file,
+                                         diff_severity=RESULT_SEVERITY.NORMAL,
+                                         diff_message="Inconsistency found."):
+                """
+                Processes the executable's output as a corrected file.
+
+                :param output:
+                    The output of the program. This can be either a single
+                    string or a sequence of strings.
+                :param filename:
+                    The filename of the file currently being corrected.
+                :param file:
+                    The contents of the file currently being corrected.
+                :param diff_severity:
+                    The severity to use for generating results.
+                :param diff_message:
+                    The message to use for generating results.
+                :return:
+                    An iterator returning results containing patches for the
+                    file to correct.
+                """
+                if isinstance(output, str):
+                    output = (output,)
+
+                for string in output:
+                    for diff in Diff.from_string_arrays(
+                                file,
+                                string.splitlines(keepends=True)).split_diff():
```

Comment:
Why is this `splitlines()` done ? Isn't output already a list of lines, making `string` a single line ?